src/workerd/jsg/util.c++

```diff
@@ -315,54 +315,39 @@ DecodedException decodeTunneledException(v8::Isolate* isolate,
   result.isFromRemote = tunneledInfo.isFromRemote;
   result.isDurableObjectReset = tunneledInfo.isDurableObjectReset;
 
-  // TODO(cleanup): This code has gotten pretty ugly, could probably use some refactoring.
-  if (errorType.startsWith("Error")) {
-    auto message = appMessage(errorType.slice(strlen("Error")));
-    result.handle = v8::Exception::Error(v8Str(isolate, message));
-  } else if (errorType.startsWith("RangeError")) {
-    auto message = appMessage(errorType.slice(strlen("RangeError")));
-    result.handle = v8::Exception::RangeError(v8Str(isolate, message));
-  } else if (errorType.startsWith("TypeError")) {
-    auto message = appMessage(errorType.slice(strlen("TypeError")));
-    result.handle = v8::Exception::TypeError(v8Str(isolate, message));
-  } else if (errorType.startsWith("SyntaxError")) {
-    auto message = appMessage(errorType.slice(strlen("SyntaxError")));
-    result.handle = v8::Exception::SyntaxError(v8Str(isolate, message));
-  } else if (errorType.startsWith("ReferenceError")) {
-    auto message = appMessage(errorType.slice(strlen("ReferenceError")));
-    result.handle = v8::Exception::ReferenceError(v8Str(isolate, message));
-  } else if (errorType.startsWith("CompileError")) {
-    auto message = appMessage(errorType.slice(strlen("CompileError")));
-    result.handle = v8::Exception::WasmCompileError(v8Str(isolate, message));
-  } else if (errorType.startsWith("LinkError")) {
-    auto message = appMessage(errorType.slice(strlen("LinkError")));
-    result.handle = v8::Exception::WasmCompileError(v8Str(isolate, message));
-  } else if (errorType.startsWith("RuntimeError")) {
-    auto message = appMessage(errorType.slice(strlen("RuntimeError")));
-    result.handle = v8::Exception::WasmCompileError(v8Str(isolate, message));
-  } else if (errorType.startsWith("DOMException")) {
-    errorType = errorType.slice(strlen("DOMException"));
+#define HANDLE_V8_ERROR(error_name, error_type) \
+  if (errorType.startsWith(error_name)) { \
+    auto message = appMessage(errorType.slice(strlen(error_name))); \
+    result.handle = v8::Exception::error_type(v8Str(isolate, message)); \
+    break; \
+  }
+
+  do {
```

Comment:
Nit: The do/while wrapper around this likely isn't necessary?

Reply:
The do/while and the associated break statements are used to avoid needing else if and to only have one internal error clause, control flow would be difficult to handle otherwise